pm/recipient_test.go

```diff
@@ -0,0 +1,851 @@
+package pm
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"math/big"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func newRecipientFixture(t *testing.T) (ethcommon.Address, *stubBroker, *stubValidator, *stubTicketStore, *big.Int, *big.Int, []byte) {
+	return randAddressOrFatal(t), newStubBroker(), &stubValidator{}, newStubTicketStore(), big.NewInt(100), big.NewInt(100), []byte("foo")
+}
+
+func newTicket(sender ethcommon.Address, params *TicketParams, senderNonce uint64) *Ticket {
+	return &Ticket{
+		Recipient:         ethcommon.Address{},
+		Sender:            sender,
+		FaceValue:         params.FaceValue,
+		WinProb:           params.WinProb,
+		SenderNonce:       senderNonce,
+		RecipientRandHash: params.RecipientRandHash,
+	}
+}
+
+func genRecipientRand(sender ethcommon.Address, secret [32]byte, seed *big.Int) *big.Int {
+	h := hmac.New(sha256.New, secret[:])
+	h.Write(append(seed.Bytes(), sender.Bytes()...))
+	return new(big.Int).SetBytes(h.Sum(nil))
+}
+
+func TestReceiveTicket_InvalidRecipientRand_InvalidSeed(t *testing.T) {
+	sender, b, v, ts, faceValue, winProb, sig := newRecipientFixture(t)
+
+	r, err := NewRecipient(b, v, ts, faceValue, winProb)
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
These few lines of creating a new recipient repeat in several tests - perhaps it's nice to extract it into a helper function like `newRecipientOrFatal(t)` and have it just return the recipient?

Reply:
Actually, you might be able to just add this logic to `newRecipientFixture`